### ISSUE TITLE
Serial Library

### ISSUE THREAD

**Issue by bkubicek** [bkubicek/Marlin#1]
The arduino serial library is non-optimized. We should transit to a custom plain atmel code without arduino dependency.

**Comment by bradleyf** [comment]
Agree.  That's one of the first things that acts up once any code is introduced with any delay
